Added motion models for Husky, AUV and USV

majcote opened the pull request:
With regards to issue #1

majcote committed:
Added motion models for Husky, AUV and USV

alexsb92 commented:
Talk about short and sweet. The only thing I could think of is adding a quick example for each? That way you might save people interested in checking them out the time to write a little demo themselves.

RobWks commented:
Looks great, I liked that you referenced the actual Clearpath robots in each case. I agree with Alex about the example, even just something that plotted the motion with a given input,  a keyboard input would be interesting. The only issue I noticed is you are dividing the thrust by the mass to get the acceleration, but you are treating it as though it is velocity when you multiply it by the time step to get your displacement? This occurs in both the USV and AUV models. I think you're going to need to have the previous velocity state as an input. Might be a bit much, but It'd be cool if you added some simple drag to the AUV, even if it's just assumed to be a sphere or something so that you could see it slow down without a thruster being active.

majcote committed:
-Added motion_model_example,
- Added a rough calculation to take into account the effect of drag  for the AUV and SUV model
- The AUV and USV model's now pass previous velocity information

majcote commented:
Thanks for the feedback, Good catch on the acceleration!

Added a few examples as suggested.

Also added in some really rough drag calculations for the USV and AUV, not perfect, but it does slow the robot down in the absence of thrust, and it does reach a max speed so the robot doesn't accelerate indefinitely.
